src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -123,18 +122,22 @@ public ImmutableArray<ArgumentProvider> GetArgumentProviders(Workspace workspace
             return _argumentProviders;
         }
 
-        public abstract int GetExpansionFunction(IXMLDOMNode xmlFunctionNode, string bstrFieldName, out IVsExpansionFunction pFunc);
-        protected abstract ITrackingSpan InsertEmptyCommentAndGetEndPositionTrackingSpan();
+        public abstract int GetExpansionFunction(IXMLDOMNode xmlFunctionNode, string bstrFieldName, out IVsExpansionFunction? pFunc);
+        protected abstract ITrackingSpan? InsertEmptyCommentAndGetEndPositionTrackingSpan();
         internal abstract Document AddImports(Document document, int position, XElement snippetNode, bool placeSystemNamespaceFirst, bool allowInHiddenRegions, CancellationToken cancellationToken);
 
         public int FormatSpan(IVsTextLines pBuffer, VsTextSpan[] tsInSurfaceBuffer)
         {
             AssertIsForeground();
 
+            if (ExpansionSession == null)
+            {
+                return VSConstants.E_FAIL;
+            }
+
             // If this is a manually-constructed snippet for a full method call, avoid formatting the snippet since
-            // doing so will disrupt signature help. Check '_state._methods' instead of '_state.IsFullMethodCallSnippet'
-            // because '_state._expansionSession' is not initialized at this point.
-            if (!_state._methods.IsDefault)
+            // doing so will disrupt signature help.
+            if (_state.IsFullMethodCallSnippet)
```

Comment:
❔ Was this changed as part of the wrong commit?

Reply:
➡️ Yes, there were a few small cases of this

Reply:
I think I did screw up the commit rebasing at one point.

Reply:
Oh, for this one, it's intentionally part of the null annotation commit: the comment stated that we had to check _methods because _expansionSession isn't initialized, but the explicit null check being added means the comment is now stale. The null check was added because the null annotation revealed that if it was null, we'd end up throwing at line 177.